mautic/mautic
# [wip] improve speed of phpunit tests

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

There are 260 seconds of `sleep` in the tests extending the `AbstractCampaignCommand` class, e.g. [here](https://github.com/mautic/mautic/blob/0abd6b79738f7cf096b435d8f87ea998617e17c1/app/bundles/CampaignBundle/Tests/Command/ValidateEventCommandTest.php#L29).
The sleep makes sense to ensure the test is executed past the scheduled date, but 20 seconds per step seems too much.

the total run time is currently around 10 minutes, so this change should (if the tests actually pass) speed ut the tests significantly

Marking this PR as draft, as I may need to run the pipeline multiple times to ensure the timings are realistic



#### Steps to test this PR:

If no tests are failing while rerunning this pipeline multiple times, this should be good.






<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10859"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
